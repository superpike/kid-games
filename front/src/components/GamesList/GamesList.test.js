import React from 'react';
import { render } from '@testing-library/react';
import { GameList } from './index';
// query utilities:
// import {
//     getByLabelText,
//     getByText,
//     getByTestId,
//     getByDisplayValue,
//     // Tip: all queries are also exposed on an object
//     // called "queries" which you could import here as well
//     wait,
// } from '@testing-library/dom'
// adds special assertions like toHaveTextContent
import '@testing-library/jest-dom/extend-expect';

describe('Testing gamesList', () => {
  const games = [
    {
      name: 'qwert',
      id: 1,
    },
    {
      name: 'qwert2',
      id: 2,
    },
  ];
  describe('Is label', () => {
    test('should be label for list', () => {
      const { container } = render(
        <GameList
          games={games}
          activeGame={{ ...games[0] }}
          isFavorite={false}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
  describe('List of the games', () => {
    test('should be chosen an active game', () => {
      const { container } = render(
        <GameList
          games={games}
          activeGame={{ ...games[1] }}
          isFavorite={false}
        />
      );
      expect(container).toMatchSnapshot();
    });
    test('after click should appears a list', async () => {
      const { container, getByText } = render(
        <GameList
          games={[
            {
              name: 'qwert',
              id: 1,
            },
            {
              name: 'qwert2',
              id: 2,
            },
          ]}
          activeGame={{ name: 'qwert2', id: 2 }}
          isFavorite={false}
        />
      );
      const button = container.querySelector('.fa-chevron-down')
        .parentElement;
      button.click();
      expect(getByText('qwert')).toBeInTheDocument();
      expect(getByText('qwert2')).toBeInTheDocument();
    });
  });
  describe('Favorites', () => {
    test('should render favorite icon', () => {
      const { container } = render(
        <GameList
          games={games}
          activeGame={{ ...games[0] }}
          isFavorite
        />
      );
      expect(container).toMatchSnapshot();
    });
    test('should render non-favorite icon', () => {
      const { container } = render(
        <GameList
          games={games}
          activeGame={{ ...games[0] }}
          isFavorite={false}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
