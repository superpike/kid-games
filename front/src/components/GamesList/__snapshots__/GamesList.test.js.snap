// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing gamesList Favorites should render favorite icon 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="gamesList"
      >
        games for you
      </label>
      <div
        class="inputWrapper"
      >
        <input
          class="InputElement"
          data="[object Object],[object Object]"
          error=""
          name="gamesList"
          selecttext="Select from the list:"
          value="qwert"
        />
        <div
          class="icon"
          role="button"
          tabindex="0"
        >
          <i
            class="fas fa-chevron-down"
          />
        </div>
      </div>
    </div>
    <div
      class="favorite"
    >
      <i
        class="fas fa-star"
      />
    </div>
  </div>
</div>
`;

exports[`Testing gamesList Favorites should render non-favorite icon 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="gamesList"
      >
        games for you
      </label>
      <div
        class="inputWrapper"
      >
        <input
          class="InputElement"
          data="[object Object],[object Object]"
          error=""
          name="gamesList"
          selecttext="Select from the list:"
          value="qwert"
        />
        <div
          class="icon"
          role="button"
          tabindex="0"
        >
          <i
            class="fas fa-chevron-down"
          />
        </div>
      </div>
    </div>
    <div
      class="favorite"
    >
      <i
        class="far fa-star"
      />
    </div>
  </div>
</div>
`;

exports[`Testing gamesList Is label should be label for list 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="gamesList"
      >
        games for you
      </label>
      <div
        class="inputWrapper"
      >
        <input
          class="InputElement"
          data="[object Object],[object Object]"
          error=""
          name="gamesList"
          selecttext="Select from the list:"
          value="qwert"
        />
        <div
          class="icon"
          role="button"
          tabindex="0"
        >
          <i
            class="fas fa-chevron-down"
          />
        </div>
      </div>
    </div>
    <div
      class="favorite"
    >
      <i
        class="far fa-star"
      />
    </div>
  </div>
</div>
`;

exports[`Testing gamesList List of the games should be chosen an active game 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="wrapper"
    >
      <label
        class="label"
        for="gamesList"
      >
        games for you
      </label>
      <div
        class="inputWrapper"
      >
        <input
          class="InputElement"
          data="[object Object],[object Object]"
          error=""
          name="gamesList"
          selecttext="Select from the list:"
          value="qwert2"
        />
        <div
          class="icon"
          role="button"
          tabindex="0"
        >
          <i
            class="fas fa-chevron-down"
          />
        </div>
      </div>
    </div>
    <div
      class="favorite"
    >
      <i
        class="far fa-star"
      />
    </div>
  </div>
</div>
`;
